shub/utils.py

```diff
@@ -258,16 +258,20 @@ def build_and_deploy_egg(project, endpoint, apikey):
     _deploy_dependency_egg(project, endpoint, apikey)
 
 
-def _deploy_dependency_egg(project, endpoint, apikey):
-    name = _get_dependency_name()
-    version = pwd_version()
-    egg_name, egg_path = _get_egg_info(name)
-
+def _deploy_dependency_egg(project, endpoint, apikey, name=None, version=None, egg_info=None):
+    if name is None:
+        name = _get_dependency_name()
+    if version is None:
+        version = pwd_version()
+
+    if egg_info is None:
+        egg_info = _get_egg_info(name)
```

Comment:
nitpick: `name = name or _get_dependency_name()` and similar are a little shorter (and implement the same behaviour since we don't allow empty strings as name or version)